src/cli/lint/annotate/github/eslint.ts

```diff
@@ -0,0 +1,29 @@
+import * as GitHub from '../../../../api/github';
+import { ESLintOutput } from '../../../../cli/adapter/eslint';
+
+export const createEslintAnnotations = (eslint: ESLintOutput) => {
+  const annotations: GitHub.Annotation[] = [];
+  [...(!eslint.ok ? eslint.errors : []), ...eslint.warnings].forEach(
+    (result) => {
+      result.messages.forEach((message) => {
+        // Annotations only support start_column and end_column on the same line.
+        const column = message.line === message.endLine && message.column;
+        const endColumn =
+          message.line === message.endLine && (message.endColumn || column);
+
+        annotations.push({
+          annotation_level: message.severity === 2 ? 'failure' : 'warning',
+          start_line: message.line || 1,
+          end_line: message.endLine || message.line || 1,
+          ...(column && { start_column: column }),
+          ...(endColumn && { end_column: endColumn }),
+          message: message.message,
+          path: result.filePath,
+          title: `Eslint${message.ruleId ? ` - ${message.ruleId}` : ''}`,
```

Comment:
I'd have to see how this is rendered out, but I'm wondering if this would be better as

```suggestion
          title: `ESLint${message.ruleId ? ` (${message.ruleId})` : ''}`,
```

Reply:
Looks like this:

![image](https://user-images.githubusercontent.com/18017094/139575305-12694cd1-1054-4a38-92a8-06643a648a75.png)

I'm indifferent. But I'll leave it as it is committed right now - with brackets